detectable/src/main/java/com/synopsys/integration/detectable/detectables/git/GitUrlParser.java

```diff
@@ -0,0 +1,42 @@
+package com.synopsys.integration.detectable.detectables.git;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class GitUrlParser {
+    // Parses urls such as: https://github.com/blackducksoftware/synopsys-detect
+    public String getRepoName(String remoteUrlString) throws MalformedURLException {
+        String[] pieces = remoteUrlString.split("[/:]");
+        if (pieces.length >= 2) {
+            String organization = pieces[pieces.length - 2];
+            String repo = pieces[pieces.length - 1];
+            String name = String.format("%s/%s", organization, repo);
+            return StringUtils.removeEnd(StringUtils.removeStart(name, "/"), ".git");
+        } else {
+            throw new MalformedURLException("Failed to extract repository name from url. Not logging url for security.");
+        }
+    }
+
+    /**
+     * Strip out credentials if the string is a URI and contains credentials.
+     * @param remoteUrlString
+     * @return sanitized URI or original string
+     */
+    public String removeCredentialsFromUri(String remoteUrlString) {
```

Comment:
This method is new.

Reply:
Could we add to the method description the expected format for a URL that includes the token? In the original ticket the example they have is: [https://gitlab-ci-token:[tokentext]@sig-gitlab.internal.synopsys.com/reponame](https://gitlab-ci-token:%5Btokentext%5D@sig-gitlab.internal.synopsys.com/reponame)